devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -233,6 +233,8 @@
     @Override
     public void execute() throws MojoFailureException, MojoExecutionException {
 
+        //we always want to compile if needed, so if it is run from the parent it will compile dependent projects
+        handleAutoCompile();
```

Comment:
I would have put that one below the line that enforces the Maven version?

Reply:
Addressed in #6953